synapse/handlers/room.py

```diff
@@ -1444,7 +1444,7 @@ async def get_new_events(
     def get_current_key(self) -> RoomStreamToken:
         return self.store.get_room_max_token()
 
-    def get_current_key_for_room(self, room_id: str) -> Awaitable[str]:
+    def get_current_key_for_room(self, room_id: str) -> Awaitable[RoomStreamToken]:
         return self.store.get_room_events_max_id(room_id)
```

Comment:
Previously, `get_current_key_for_room` was unused along with `get_room_events_max_id`.

It seemed almost exactly what I was looking for so I repurposed both functions with proper `RoomStreamToken` types instead of returning the string version that was interpolated together.